### ISSUE TITLE
Server Error 500 when attempting to open playlist

### ISSUE THREAD

**Issue by cosmoneer** [tubearchivist/tubearchivist#223]
Hi,

I was able to successfully download videos and subscribe to a playlist, but I receive a "Server Error (500)" when I click on it.  The playlist is from my own YouTube account and is accessible, but is unlisted.  Both videos referenced by the playlist exist and are playable in the Archivist inventory.

Here is the YouTube playlist URL:

https://www.youtube.com/playlist?list=PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb

Here is the error log entry for the error 500:

[pid: 33|app: 0|req: 4574/4574] 10.0.0.120 () {44 vars in 987 bytes} [Tue Apr 12 13:30:09 2022] GET /playlist/PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb/ => generated 145 bytes in 156 msecs (HTTP/1.1 500) 7 headers in 233 bytes (1 switches on core 0)

Here are the log entries from when the playlist was added:

https://www.youtube.com/playlist?list=PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb
[2022-04-12 12:22:24,595: INFO/MainProcess] Task home.tasks.subscribe_to[0735d0f5-53ad-41cb-b3c4-8b541a5a2148] received
[2022-04-12 12:22:24,597: WARNING/ForkPoolWorker-8] processing: https://www.youtube.com/playlist?list=PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb

[2022-04-12 12:22:24,597: WARNING/ForkPoolWorker-8] ParseResult(scheme='https', netloc='www.youtube.com', path='/playlist', params='', query='list=PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb', fragment='')

[2022-04-12 12:22:24,667: WARNING/ForkPoolWorker-8] PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb: get metadata from es

[2022-04-12 12:22:24,674: WARNING/ForkPoolWorker-8] {"_index":"ta_playlist","_type":"_doc","_id":"PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb","found":false}

[2022-04-12 12:22:24,675: WARNING/ForkPoolWorker-8] PLjUjOrK_Y-_1sgE94UPvqA5OqKXPDNYwb: get metadata from youtube

[2022-04-12 12:22:25,107: WARNING/ForkPoolWorker-8] downloading 0 playlist artwork

[2022-04-12 12:22:25,386: WARNING/ForkPoolWorker-8] UCXz8cDYJB9N53oIUjcJB2dw: get metadata from es

[2022-04-12 12:22:25,392: WARNING/ForkPoolWorker-8] {"_index":"ta_channel","_type":"_doc","_id":"UCXz8cDYJB9N53oIUjcJB2dw","found":false}

[2022-04-12 12:22:25,392: WARNING/ForkPoolWorker-8] UCXz8cDYJB9N53oIUjcJB2dw: scrape channel data from youtube

[2022-04-12 12:22:25,560: WARNING/ForkPoolWorker-8] downloading 1 channel artwork

[pid: 33|app: 0|req: 4508/4508] 10.0.0.120 () {52 vars in 1102 bytes} [Tue Apr 12 12:22:24 2022] POST /playlist/ => generated 0 bytes in 1010 msecs (HTTP/1.1 302) 8 headers in 252 bytes (1 switches on core 0)
[pid: 33|app: 0|req: 4509/4509] 10.0.0.120 () {46 vars in 948 bytes} [Tue Apr 12 12:22:25 2022] GET /playlist/ => generated 6755 bytes in 23 msecs (HTTP/1.1 200) 8 headers in 398 bytes (1 switches on core 0)
[pid: 33|app: 0|req: 4510/4510] 10.0.0.120 () {42 vars in 750 bytes} [Tue Apr 12 12:22:25 2022] GET /progress/ => generated 135 bytes in 8 msecs (HTTP/1.1 200) 7 headers in 221 bytes (1 switches on core 0)
[2022-04-12 12:22:25,790: WARNING/ForkPoolWorker-8] downloading 1 playlist artwork

[2022-04-12 12:22:25,866: INFO/ForkPoolWorker-8] Task home.tasks.subscribe_to[0735d0f5-53ad-41cb-b3c4-8b541a5a2148] succeeded in 1.2695909890171606s: None

**Comment by cosmoneer** [comment]
As a test, I added an existing playlist from a channel whose video I had already archived.  Tube Archivist had no issues showing the playlist for this channel when I clicked on it.

I then went to the channels list and did not find a channel listed for my account where I had subscribed my playlist.  I suspect this is the issue, in that the server cannot display the playlist page without a channel entry.  With this in mind, I found a video I had published to my account and downloaded it, which caused Archivist to create a channel for my account.

I then returned to the playlists page and was able to open the playlist from my YouTube account.

Maybe Tube Archivist needs to check to see if a channel entry exists for the playlist being subscribed to and create it if none exists?

Anyway, thanks for the awesome app!
